Realm/RLMGeospatial.h

```diff
@@ -0,0 +1,228 @@
+ inside the outer `Polygon`.
+ A `Polygon` describes a shape conformed by at least three segments, where the last and the first `RLMGeospatialPoint`
+ must be the same to indicate a closed polygon (meaning you need at least 4 points to define a polygon).
+ Inner holes in a `RLMGeospatialPolygon` must  be entirely inside the outer ring
+
+ A `hole` has the following restrictions:
+ - Holes may not cross, i.e. the boundary of a hole 
+ may not intersect both the interior and the exterior of any other
+   hole.
+ - Holes may not share edges, i.e. if a hole contains and edge AB, the no other hole may contain it.
+ - Holes may share vertices, however no vertex may appear twice in a single hole.
+ - No hole may be empty.
+ - Only one nesting is allowed.
+
+ @warning This class cannot be persisted and can only be use within a geospatial `geoWithin` query.
+ */
+@interface RLMGeospatialPolygon : RLMGeospatial
+/// The polygon's external (outer) ring.
+@property (readonly, strong) NSArray<RLMGeospatialPoint *> *outerRing;
+/// The holes (if any) in the polygon.
+@property (readonly, strong, nullable) NSArray<NSArray<RLMGeospatialPoint *> *> *holes;
+
+/**
+Initialize a `RLMGeospatialPolygon`, with its outer rings and holes (if any).
+
+Returns `nil` if the `RLMGeospatialPoints` representing a polygon (outer ring or holes), don't have at least 4 points.
+Returns `nil` if the first and the last `RLMGeospatialPoint` in a polygon are not the same.
+
+@param outerRing The polygon's external (outer) ring.
+ */
+- (nullable instancetype)initWithOuterRing:(NSArray<RLMGeospatialPoint *> *)outerRing;
+
+/**
+Initialize a `RLMGeospatialPolygon`, with its outer rings and holes (if any).
+
+Returns `nil` if the `RLMGeospatialPoints` representing a polygon (outer ring or holes), don't have at least 4 points.
+Returns `nil` if the first and the last `RLMGeospatialPoint` in a polygon are not the same.
+
+@param outerRing The polygon's external (outer) ring.
+@param holes The holes (if any) in the polygon.
+ */
+- (nullable instancetype)initWithOuterRing:(NSArray<RLMGeospatialPoint *> *)outerRing holes:(nullable NSArray<NSArray<RLMGeospatialPoint *> *> *)holes;
+@end
+
+/**
+ This structure is a helper to represent/convert a distance. It can be used in geospatial
+ queries like those represented by a `RLMGeospatialCircle`
+
+ - warning: This structure cannot be persisted and can only be used to build other geospatial shapes
+ */
+@interface RLMDistance : RLMGeospatial
+/// The distance in radians.
+@property (readonly) double radians;
+
+/**
+Constructs a `Distance`.
+
+Returns `nil` if the value is lower than 0, because we cannot construct negative distances.
+@param kilometers Distance in kilometers.
+@returns A value that represents the provided distance in radians.
+ */
++ (nullable instancetype)initFromKilometers:(double)kilometers NS_SWIFT_NAME(fromKilometers(kilometers:));
```

Comment:
A class function named `init...` doesn't really make any sense; that would be a method which initializes the class itself, not something that returns an instance of the class. A class method which returns a RLMDistance would be `distanceFromKilometers:`.

In Swift the argument label shouldn't be repeated, and there isn't any obvious reason for it to be a factory function rather than an initializer.

Reply:
Actually, modeling the API off [Duration](https://developer.apple.com/documentation/swift/duration) might make sense, which would involve static factory functions like `static func kilometers(_ kilometers: Double) -> Distance` resulting in `GeoCircle(center: (0, 0), radius: .miles(100))`.